src/main/java/com/metamx/cache/CaffeineCache.java

```diff
@@ -45,15 +49,31 @@
 
   public static CaffeineCache create(final CaffeineCacheConfig config)
   {
-    Caffeine<Object, Object> builder = Caffeine.newBuilder().recordStats();
+    return create(config, null);
+  }
+
+  // Used in testing
+  public static CaffeineCache create(final CaffeineCacheConfig config, @Nullable final Executor executor)
+  {
+    Caffeine builder = Caffeine.newBuilder().recordStats();
     if (config.getExpiration() >= 0) {
       builder = builder
           .expireAfterAccess(config.getExpiration(), TimeUnit.MILLISECONDS);
     }
     if (config.getMaxSize() >= 0) {
       builder = builder
-          .maximumSize(config.getMaxSize())
-          .maximumWeight(config.getMaxSize());
+          .maximumWeight(config.getMaxSize())
+          .weigher(new Weigher<NamedKey, byte[]>()
```

Comment:
Sure, unfortunately the use case here requires features added conditionally based on configuration.